compat/inet_ntop.cc

```diff
@@ -163,23 +163,22 @@ inet_ntop6(const u_char *src, char *dst, size_t size)
      * Keep this in mind if you think this function should have been coded
      * to use pointer overlays.  All the world's not a VAX.
      */
-    char tmp[sizeof "ffff:ffff:ffff:ffff:ffff:ffff:255.255.255.255"], *tp;
-    struct { int base, len; } best, cur;
-    u_int words[NS_IN6ADDRSZ / NS_INT16SZ];
-    int i;
+    char tmp[sizeof "ffff:ffff:ffff:ffff:ffff:ffff:255.255.255.255"];
+    struct match_ {
+        int base = -1;
+        int len = 0;
+    };
+    match_ best;
+    match_ cur;
+    u_int words[NS_IN6ADDRSZ / NS_INT16SZ] = {};
 
     /*
      * Preprocess:
      *  Copy the input (bytewise) array into a wordwise array.
      *  Find the longest run of 0x00's in src[] for :: shorthanding.
      */
-    memset(words, '\0', sizeof words);
-    for (i = 0; i < NS_IN6ADDRSZ; i++)
+    for (int i = 0; i < NS_IN6ADDRSZ; ++i)
         words[i / 2] |= (src[i] << ((1 - (i % 2)) << 3));
-    best.base = -1;
-    best.len = 0;
-    cur.base = -1;
-    cur.len = 0;
     for (i = 0; i < (NS_IN6ADDRSZ / NS_INT16SZ); i++) {
```

Comment:
After moving the earlier function-scope declaration into previous for-loop scope this ```i``` is undefined.

Same for other loop below.

Reply:
To be more precise, `i` is undeclared (not undefined). More `int`s are needed for those loops.

More importantly, we need to at least compile this code before merging this PR -- same problem as in https://github.com/squid-cache/squid/pull/1079#pullrequestreview-1026897488 :-(

Reply:
> More importantly, we need to at least compile this code before merging this PR -- same problem as in [#1079 (review)](https://github.com/squid-cache/squid/pull/1079#pullrequestreview-1026897488) :-(

Right :( I suppose that systems lacking this function are quite rare, especially among the ones we support.
Might be worthwhile to actually run a test to see what we could drop from compat/

Rechecking both PRs and ensuring they can build. Apologies for the noise.

Reply:
tested with CPPFLAGS=-D_ARPA_INET_H_ to prevent the system header from being included. Now passes build. updating pr

Reply:
Confirmed tested and working. resolving